src/lib/coda_lib/coda_lib.ml

```diff
@@ -14,21 +15,59 @@ open Otp_lib
 (* used in error below to allow pattern-match against error *)
 let refused_answer_query_string = "Refused to answer_query"
 
-module Make (Inputs : Intf.Inputs) = struct
-  open Auxiliary_database
-  open Inputs
-  module Ledger_transfer = Ledger_transfer.Make (Ledger) (Ledger.Db)
-  module Subscriptions = Coda_subscriptions.Make (Inputs)
+open Auxiliary_database
+module Ledger_transfer = Ledger_transfer.Make (Ledger) (Ledger.Db)
+module Subscriptions = Coda_subscriptions
 
+module Config = struct
+  (** If ledger_db_location is None, will auto-generate a db based on a UUID *)
+  type t =
+    { logger: Logger.t
+    ; trust_system: Trust_system.t
+    ; verifier: Verifier.t
+    ; prover: Prover.t
+    ; initial_propose_keypairs: Keypair.Set.t
+    ; snark_worker_key: Public_key.Compressed.Stable.V1.t option
+    ; net_config: Coda_networking.Config.t
+    ; snark_pool_disk_location: string
+    ; wallets_disk_location: string
+    ; ledger_db_location: string option
+    ; transition_frontier_location: string option
+    ; staged_ledger_transition_backup_capacity: int [@default 10]
+    ; time_controller: Block_time.Controller.t
+    ; receipt_chain_database: Coda_base.Receipt_chain_database.t
+    ; transaction_database: Transaction_database.t
+    ; external_transition_database: External_transition_database.t
+    ; snark_work_fee: Currency.Fee.t
+    ; monitor: Monitor.t option
+    ; consensus_local_state: Consensus.Data.Local_state.t
+          (* TODO: Pass banlist to modules discussed in Ban Reasons issue: https://github.com/CodaProtocol/coda/issues/852 *)
+    }
+  [@@deriving make]
+end
+
+module Make
+    (Work_selector : Work_selector.Intf.S
```

Comment:
Should selecting a `Work_selector` be a compiled time variable. I feel like it should be.

Reply:
I think we want to leave it runtime, but I'm currently removing it as a functor input in favor of a first class module. Just doing that outside of this PR to leave the scope small.